src/idl_gen_dart.cpp

```diff
@@ -449,13 +449,22 @@ class DartGenerator : public BaseGenerator {
     code += "  final " + _kFb + ".BufferContext _bc;\n";
     code += "  final int _bcOffset;\n\n";
 
-    GenImplementationGetters(struct_def, &code);
+    std::vector<std::pair<int, FieldDef*>> non_deprecated_fields;
+    for (auto it = struct_def.fields.vec.begin();
+           it != struct_def.fields.vec.end(); ++it) {
+        auto &field = **it;
+        if (field.deprecated) continue;
+        auto offset = it - struct_def.fields.vec.begin();
+        non_deprecated_fields.push_back(std::make_pair(offset, &field));
```

Comment:
## Breaks windows Build

I have some reason to believe these two lines [#457-458](https://github.com/google/flatbuffers/blob/316d7c20898ce95eacbbca734448538b66d6785f/src/idl_gen_dart.cpp#L457-L458) have broken the windows build.
I dont know anything about dart but, when I comment out these two specific  lines I'm able to build the entire project (albeit minus the proper fix intended by this PR)

Here is the specific error message:
```
[build] cl : Command line warning D9025 : overriding '/W3' with '/W4'
[build] C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include\utility(239): error C2220: warning treated as error - no 'object' file generated
[build] src\idl_gen_dart.cpp(458): note: see reference to function template instantiation 'std::pair<int,flatbuffers::FieldDef *>::pair<__int64,T*,0>(std::pair<__int64,T *> &&) noexcept' being compiled
[build]         with
[build]         [
[build]             T=flatbuffers::FieldDef
[build]         ]
```
I'm using vscode on windows, by the way.
@dnfield can you confirm windows build still works?

Reply:
by the way, I probably should mention that I dont know anything about this particular file, or what the significance of those lines are. I work mainly on golang code. I was just testing building on windows for my upcoming golang PR and noticed this error

Reply:
Yes, this was also pointed out above.  I could submit a patch to fix it but I don't have mavc to test with easily.

Reply:
oh sorry didn't even see that comment.

anyhow, perhaps the more important question is why didn't the ci suite pick this up?

Reply:
Looks like ci is using vs 2015.  Maybe this is a 2017 or 2019 bug?

Reply:
ah I see! I'm using 2017